### PR TITLE
Fixes #42: Configurable settings from the json config file.

### LINKED ISSUE #42: Allow preconfiguring server settings in the admin config file

### PR DESCRIPTION
**Pull request by n1k0**
This patch updates the admin JSON configuration file to allow configuring default server settings. Note that this change breaks backward compatibility and require upgraders to update their config file. 

r=? @leplatrem @magopian

### CODE REVIEW

**Comment by n1k0** [comment]
I need to ensure custom settings entered in the dedicated form are persisted locally; right now they're overridden by config settings every time the page is reloaded, which is wrong.

File: package.json [hunk 1]
```diff
@@ -58,6 +58,7 @@
     "fake-indexeddb": "1.0.3",
     "file-loader": "^0.8.5",
     "gh-pages": "^0.11.0",
+    "history": "^1.17.0",
```
Comment (leplatrem) [inline package.json]: just curious, what is that ?

**Comment by leplatrem** [comment]
LGTM :)

File: package.json (hunk 1 above)
Comment (n1k0) [inline package.json]: A peer dependency which was handled automatically just fine with npm 2 but which breaks with npm 3; declaring it here makes it work in both versions :)

**Pull request merged**